File: ios/RatingControlManager.m
//
//  RatingControlManager.m
//  CounterApp
//
//  Created by evhive on 09/04/19.
//  Copyright © 2019 Facebook. All rights reserved.
//

#import "React/RCTViewManager.h"
//
// the component will be exposed as "CounterView"
// if the name ends with "Manager" it will be stripped by React Native
@interface RCT_EXTERN_MODULE(RatingControlManager, RCTViewManager)
// or, we could also rename it ourselves
//@interface RCT_EXTERN_REMAP_MODULE(CounterView, CounterViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(rating, NSNumber)

// the type could also be "RCTDirectEventBlock", but the difference is not explained anywhere
// the name must begin with lowercase "on"
RCT_EXPORT_VIEW_PROPERTY(onUpdates, RCTBubblingEventBlock)

// all NSNumber arguments are required to be marked as "nonnull"
RCT_EXTERN_METHOD(updateFromManager:(nonnull NSNumber *)node count:(nonnull NSNumber *)count)

@end

File: ios/RatingControlManager.swift
//
//  RatingControlManager.swift
//  CounterApp
//
//  Created by evhive on 09/04/19.
//  Copyright © 2019 Facebook. All rights reserved.
//

import Foundation

@available(iOS 9.0, *)
@objc(RatingControlManager)
class RatingControlManager: RCTViewManager {
  override func view() -> UIView! {
    let stackView: UIStackView! = RatingControl()
//    stackView.distribution = .fillEqually
//    stackView.alignment = .center
    return stackView
  }
  
  // this is required since RN 0.49+
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }

  func updateFromManager(_ node: NSNumber, rating: NSNumber) {
    DispatchQueue.main.async {
      let component = self.bridge.uiManager.view(
        forReactTag: node
        ) as! RatingControl
      component.update(value: rating)
    }
  }
}
